EpiResponsivePicture/GeneratorProfiles/ResizedUrlGeneratorBase.cs

```diff
@@ -1,30 +1,39 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Specialized;
+using System.IO;
 using EPiServer;
 using Forte.EpiResponsivePicture.ResizedImage;
 using Forte.EpiResponsivePicture.ResizedImage.Property;
 
 namespace Forte.EpiResponsivePicture.GeneratorProfiles;
-using CustomQueryFunc = Func<int, PictureSource, PictureProfile, FocalPoint, (string Key, string Value)>;
-using CustomQueryPredicate = Func<int, PictureSource, PictureProfile, FocalPoint, bool>;
+using CustomQueryFunc = Func<int, PictureSource, ResizedImageFormat, FocalPoint, (string Key, string Value)>;
+using CustomQueryPredicate = Func<int, PictureSource, ResizedImageFormat, FocalPoint, bool>;
 
 public abstract class ResizedUrlGeneratorBase : IResizedUrlGenerator
 {
     protected UrlBuilder Builder;
 
     private readonly Queue<(CustomQueryFunc Func, CustomQueryPredicate Predicate)> customQueryRegistrations = new();
     private readonly NameValueCollection customQueries = new();
-    public UrlBuilder GenerateUrl(string imageUrl, int width, PictureSource pictureSource, PictureProfile pictureProfile, FocalPoint focalPoint)
+    public UrlBuilder GenerateUrl(string imageUrl, int width, PictureSource pictureSource, ResizedImageFormat format, FocalPoint focalPoint)
     {
+        string imageExtension = Path.GetExtension(imageUrl);
+        if (format is ResizedImageFormat.Preserve)
+        {
+            if (imageExtension is ".tif" or ".tiff")
```

Comment:
can we merge these two ifs into one?
Also, maybe we can make a property in this class that will hold collection of extensions that will be replaced with Jpeg so that in the future it's easier to extend the list?